src/Spectre.libGaussianMixtureModelling/Matrix.h

```diff
@@ -31,29 +31,68 @@ struct Matrix
 {
     /// <summary>
     /// Constructor initializing the data based on given dimensions.
+    /// It sets all the values to zero upon creation.
     /// </summary>
     /// <param name="height">Height of the array.</param>
     /// <param name="width">Width of the array.</param>
     Matrix(unsigned height, unsigned width)
     {
+        unsigned length = height * width;
         data = new Type*[height];
         if (height > 0)
         {
-            data[0] = new Type[width * height]; // allocate data once
+            data[0] = new Type[length]; // allocate data once
             for (unsigned i = 1; i < height; ++i)
                 data[i] = data[0] + i * width; // assign pointers
         }
-        memset(data[0], 0, width * height * sizeof(Type));
+        memset(data[0], 0, length * sizeof(Type));
     }
 
+    /// <summary>
```

Comment:
If this class gets extended even a bit more, switch to `cv::Mat` or just consider using Eigen.